### ISSUE TITLE
Reveal RC01 Testing Feedback

### ISSUE THREAD

**Issue by craigappl** [OpenSRP/opensrp-client-reveal#45]
### Spray Form Updates

- [x] Options on Spray Status need a space after the comma. It's currently "Sprayble,Sprayed" and we should have "Sprayable, Sprayed"
- [x] The skip logic on the first field doesn't hide all other fields if it is changed. So, touch Residential Structure, Sprayble, Not Sprayed, then go back and change the Type of Structure to Non-Residential Structure. All items that were displayed were not hidden again.
- [x] The text for Number of Eligible Rooms and Total Number of Nets Hanging go off the screen. Can we make the size of that text smaller?
- [x] "Is a mop-up activity" should be "Is this a mop-up activity?"

### Main Page
- [x] When you pull up a card, you can touch the white area in the card and it still activates points on the map. You can also, touch the white area within the card and drag your finger to move the map around. This shouldn't be the case. The white area should block user interactions from the underlying map.
- [x] When you touch a point that is not yellow, the card pops up and you can't see the point anymore. I recommend centering the view so that the point you clicked moves up to the area above the card so it's clear which one you clicked on. This should be the same behavior after you save a spray form.

### Spray Cards
- [x] The word residential should be capitalized on the card "Residential"
- [x] The "Change spray status" button should be capitalized "Change Spray Status"
- [x] Green points should display capitalized statements "Sprayable, Sprayed" instead of "Sprayable, sprayed"
- [x] Red points should display capitalized statements "Sprayable, Not Sprayed" instead of "Sprayable, not sprayed"
- [ ] ~~Change Spray Status forms don't seem to be saving ([link to QA test case](https://docs.google.com/document/d/1CCTXq_v2zBKxvi4ogfWNee3fuYZvcQzfaexF7hYK1Dk/edit#heading=h.u62xcr9yznhs))~~

### General
- [ ] ~~App doesn't work without location services enabled. We should support the app working without requiring the location services to be running. If they are not running, they should only ask to be enabled if the my location button is pushed.~~ Moved to another version
- [ ] ~~The new structure form asks for permissions to use "Phone" I'm not sure where this comes from, but it isn't intuitive that the user will have to give permissions to make a call in the event that they are adding a point.~~ This is due to side loading the app. When distributed, it will ask for these items in the beginning.

### Need input from Roger
@rowo , can you review the following:
- [x] When a user touches the add point button, it pops up a map of the operational area, but it doesn't show the existing points on the map. Should we show these points so they have a reference? **RW: Yes, apologies that wasn't clear on the mockups. The points should show in their original state (with the colors) and the marker as that pin works well to differentiate.**
- [x] When a user adds a point and saves it, the point is added, but the point there but it's not highlighted in orange. Should the point that was just added be highlighted in orange? **RW: Yes, and since we don't have that bottom popup for a selected no-status point, it'll need to be tapped again to fill in the details**
- [x] The Register Structure form asks "Does this structure have a name?" This might get a yes/no response. I think it would be better to have the question label be "Name of Structure" **RW: agreed, though this requirement came from elsewhere as the mockups said "Structure name" but "Name of structure" is probably better as it leads with the pertinent info**

**Comment by mlberg** [comment]
When we add a new point, I think it's critical we keep the existing structures visible.  The reason being is we want to avoid them getting confused and adding the same structure again.

The orange circle works well.  I'm not sure if that's the best visible way to represent selection.  Would be good to get @rowo opinion here.

The cards look really great! I do think it would be nice if the point selected centers slightly above the card.  Right now it often gets hidden behind the card for me so I can't see what cards selected.  Again, key here is making it super simple to see what structure you clicked on before you edit especially if homes are in close proximity.

All in all though.  The app looks and functions amazingly well.  Really blown away!

**Comment by githengi** [comment]
> The new structure form asks for permissions to use "Phone" I'm not sure where this comes from, but it isn't intuitive that the user will have to give permissions to make a call in the event that they are adding a point.

This is because the form wants to read device information, this is the permission that is prompted by user, if they decline to grant the permission the user is informed the reason for that permission request as per below screenshot
![screenshot from 2018-12-19 17-16-22](https://user-images.githubusercontent.com/25123762/50225519-da6a3900-03b1-11e9-94a7-8143f12199c9.png)

The message that is displayed when asking for permission is specified by android and we cannot override, we can only inform a user why the permission is required if they decline to give access.

**Comment by rowo** [comment]
We should get more specific on that permission message and also follow dialog option guidelines. First, what does "not function properly" mean? Second, the permission question should probably follow the pattern shown on the image below (right side):

![image](https://user-images.githubusercontent.com/1584163/50232862-1b495a00-0380-11e9-9178-5efcd8333bf7.png)

See article here: http://babich.biz/mobile-ux-design-the-right-ways-to-ask-users-for-permissions/

Basically you are just asking for permission again on whatever is needed and the button options are DENY  |  ALLOW, not no or yes. You can still have the benefit explanation text on why, for example:

![image](https://user-images.githubusercontent.com/1584163/50232984-68c5c700-0380-11e9-872a-99fa093e658c.png)

**Comment by githengi** [comment]
The dialog options is defined by Android and cannot be overridden by application developers.
 
![screenshot from 2018-12-19 22-42-52](https://user-images.githubusercontent.com/25123762/50243898-837b5880-03df-11e9-92b8-08efd9e2d93a.png)

@rowo We can adopt the option you suggested where we have a  custom dialog before we request for permissions. 

The permission required by native form is device identification details so that the form can capture device details 

- subscriberid
- simserial
- phonenumber

Without the permission the form will not be able to capture the above infomation. 

@rowo you can can kindly design a flow that we can implement in the native form library since this permission and message is defined in native form library

**Comment by rowo** [comment]
@githengi So just to get this straight — the wording “Are you sure you want to deny this permission?” Yes or No — that’s Anroid defined and nothing from that is changeable?  And therefore what you are suggesting is a custom dialog before this appears that explains the following dialog? Question — in that initial custom dialog, can we grant a permission or is it just informational and the second one will always appear?

**Comment by githengi** [comment]
>  “Are you sure you want to deny this permission?” Yes or No

We have defined the contents of this dialog. This is the dialog that we cannot change 

![screenshot from 2018-12-20 00-41-52](https://user-images.githubusercontent.com/25123762/50249809-1b813e00-03f0-11e9-87a8-30800406e59c.png)
